include/openPMD/IO/AbstractIOHandler.hpp

```diff
@@ -56,6 +56,28 @@ class unsupported_data_error : public std::runtime_error
     virtual ~unsupported_data_error() { }
 };
 
+/**
+ * @brief Determine what items should be flushed upon Series::flush()
+ *
+ */
+enum class FlushLevel : unsigned char
+{
+    UserFlush,
+    /**
+     * Default mode, flush everything that can be flushed
+     * Does not need to uphold user-level guarantees about clearing and filling
```

Comment:
I don't understand the 2nd sentence yet:
Are you saying that this cannot check the API contract we have with the user, aka that the user must have given us valid buffers and they are ready to use at this point?

Reply:
I've clarified documentation now by introducing a concept of flush points. Difference between both modes is that UserFlush defines a flush point and ~~FlushEverything~~ InternalFlush does not.